src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvocationContext.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Shared.Collections;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Provides context for an in-flight function invocation.</summary>
+public sealed class FunctionInvocationContext
+{
+    /// <summary>
+    /// A nop function used to allow <see cref="Function"/> to be non-nullable. Default instances of
+    /// <see cref="FunctionInvocationContext"/> start with this as the target function.
```

Comment:
Why not just have accept a constructor parameter?

Reply:
To keep everything on equal footing, rather than special-casing some parameters (the ref types). And I don't want to use init for the same reasons I've outlined in the past.